admin/services/school-impersonation.service.js

```diff
@@ -0,0 +1,44 @@
+'use strict'
+const R = require('ramda')
+
+const schoolDataService = require('../services/data-access/school.data.service')
+const schoolImpersonationDfeNumberValidator = require('../lib/validator/school-impersonation-dfe-number-validator')
+
+const schoolImpersonationService = {}
+
+/**
+ * Validate and assign school to helpdesk session
+ * @param {object} user
+ * @param {string} dfeNumber
+ * @returns {Object}
+ */
+schoolImpersonationService.validateImpersonationForm = async (user, dfeNumber) => {
+  let validationError
+  validationError = schoolImpersonationDfeNumberValidator.isDfeNumberEmpty(dfeNumber)
+  if (validationError.hasError()) {
+    return validationError
+  }
+  let school
+  try {
+    school = await schoolDataService.sqlFindOneByDfeNumber(dfeNumber)
+  } catch (error) {}
+  validationError = schoolImpersonationDfeNumberValidator.isDfeNumberValid(school)
```

Comment:
Not sure what the point of this statement is?  Either school is undefined from line 21, or it was found in the db.  Why do we need a validator to detect undefined?  Just use `if (school === undefined)`